#! /bin/sh -e 
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Fixes PReP Motorola Powerstack II Utah irqs.
## DP: Patch authors: Leigh Brown <leigh@solinno.co.uk>, Sven Luther <luther@debian.org>
## DP: Upstream status: under investigation.

. $(dirname $0)/DPATCH

@DPATCH@
--- kernel-source-2.6.11/arch/ppc/platforms/prep_pci.c.orig	2005-04-08 12:54:40.000000000 +0200
+++ kernel-source-2.6.11/arch/ppc/platforms/prep_pci.c	2005-04-08 13:25:39.000000000 +0200
@@ -1246,7 +1246,13 @@
 	}
 
 	/* Setup the Winbond or Via PIB */
-	prep_pib_init();
+	/* Setup the Winbond or Via PIB - prep_pib_init() is coded for
+	 * the non-openpic case, but it breaks (at least) the Utah
+	 * (Powerstack II Pro4000), so only call it if we have an
+	 * openpic.
+	 */
+	if (have_openpic)
+		prep_pib_init();
 }
 
 static void __init
